hipe4ml/tree_handler.py

```diff
@@ -0,0 +1,269 @@
+
+    def get_sliced_df_list(self):
+        """
+        Get the list containing the slices of the orginal
+        DataFrame
+
+        Returns
+        ------------------------------------------------
+        out: list
+            List containing the slices of the orginal
+            DataFrame
+
+        """
+        return self._sliced_df_list
+
+    def apply_preselections(self, preselections, inplace=True):
+        """
+        Apply preselections to data
+
+        Parameters
+        ------------------------------------------------
+        preselection: str
+            String containing the cuts to be applied as preselection on the data contained in the original
+            tree. The string syntax is the one required in the pandas.DataFrame.query() method.
+            You can refer to variables in the environment by prefixing them with an ‘@’ character like @a + b.
+        inplace: bool
+            If True, the preselected dataframe replaces the initial dataframe. Otherwise return a copy of the
+            preselected df
+
+        Returns
+        ------------------------------------------------
+        out: pandas.DataFrame or None
+            If inplace == True return None is returned and the full DataFrame is replaced
+
+        """
+        df_pres = None
+        if inplace:
+            self._preselections = preselections
+            self._full_data_frame = self._full_data_frame.query(
+                self._preselections)
+            return df_pres
+
+        df_pres = self._full_data_frame.query(preselections)
+        return df_pres
+
+    def slice_data_frame(self, projection_variable, projection_binning, delete_original_df=False):
+        """
+        Create a list containing slices of the orginal DataFrame.
+        The original DataFrame is splitted in N sub-DataFrames following
+        the binning(projection_binning) of a given variable(projected_variable)
+
+        Parameters
+        ------------------------------------------------
+        projection_variable: str
+            Name of the variable that will be sliced in the analysis
+        projection_binning: list
+            Binning of the sliced variable should be given as a list of
+            [min, max) values for each bin
+
+        delete_original_df: bool
+            If True delete the original DataFrame. Only the
+            the slice array will be accessible in this case
+
+        """
+
+        self._projection_variable = projection_variable
+        self._projection_binning = projection_binning
+
+        self._sliced_df_list = []
+        for ibin in projection_binning:
+            bin_mask = np.logical_and(
+                self._full_data_frame[projection_variable] >= ibin[0],
+                self._full_data_frame[projection_variable] < ibin[1])
+            self._sliced_df_list.append(self._full_data_frame[bin_mask])
+        if delete_original_df:
+            self._full_data_frame = None
```

Comment:
I am not sure whether this really deletes the dataframe (if it is the case, just ignore this comment). Wouldn't be better to use 
```python
del self._full_data_frame
``` 
instead?

Reply:
I cross-checked that and I'm almost sure that assigning self._full_data_frame = None is enough to delete it!